airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py

```diff
@@ -22,15 +22,270 @@
 SOFTWARE.
 """
 
+import argparse
+import json
 import os
-
-print(os)
+from typing import Optional, Tuple, Union
 
 
 class TransformCatalog:
-    def run(self):
-        print("running catalog transform")
+    """
+To run this transformation:
+```
+python3 main_dev_transform_catalog.py \
+  --catalog integration_tests/catalog.json \
+  --out normalization/dbt-transform/models/generated/ \
+  --json json_blob \
+  --table airbytesandbox.data.one_recipe_json
+```
+    """
+
+    config: dict = {}
+
+    def run(self, args):
+        self.parse(args)
+        for catalog_file in self.config["catalog"]:
+            print(f"Processing {catalog_file}...")
+            catalog = read_json_catalog(catalog_file)
+            result = generate_dbt_model(catalog=catalog, json_col=self.config["json"], from_table=self.config["table"])
+            self.output_sql_models(result)
+
+    def parse(self, args):
+        parser = argparse.ArgumentParser(add_help=False)
+        parser.add_argument("--catalog", nargs="+", type=str, required=True, help="path to Catalog (JSON Schema) file")
+        parser.add_argument("--out", type=str, required=True, help="path to output generated DBT Models to")
+        parser.add_argument("--json", type=str, required=True, help="name of the column containing the json blob")
+        parser.add_argument("--table", type=str, required=True, help="schema and table name containing the json blob")
+        parsed_args = parser.parse_args(args)
+        self.config = {"catalog": parsed_args.catalog, "output_path": parsed_args.out, "json": parsed_args.json, "table": parsed_args.table}
+        print(self.config)
+
+    def output_sql_models(self, result: dict):
+        output = self.config["output_path"]
+        if result:
+            if not os.path.exists(output):
+                os.makedirs(output)
+            for file, sql in result.items():
+                print(f"Generating {file.lower()}.sql in {output}")
+                with open(os.path.join(output, f"{file}.sql").lower(), "w") as f:
+                    f.write(sql)
+
+
+def read_json_catalog(input_path: str) -> dict:
+    with open(input_path, "r") as file:
+        contents = file.read()
+    return json.loads(contents)
+
+
+def is_string(property_type) -> bool:
+    return property_type == "string" or "string" in property_type
+
+
+def is_integer(property_type) -> bool:
+    return property_type == "integer" or "integer" in property_type
```

Comment:
i think you'll need to add "number" to this as well. jsonschema can come as either integer or number